radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -700,22 +700,21 @@ public Result transferTokens(
 				.flatMap(mapper -> mapper.mapToParticleGroups(a))
 		);
 		final Observable<ParticleGroup> statefulParticleGroups = this.statefulMappersToParticleGroups(allActions);
-		final Observable<ParticleGroup> timestampParticle =
-			Observable.just(ParticleGroup.of(SpunParticle.up(new TimestampParticle(System.currentTimeMillis()))));
 
-		return Observable.concat(statelessParticleGroups, statefulParticleGroups, timestampParticle)
+		return Observable.concat(statelessParticleGroups, statefulParticleGroups)
 			.<List<ParticleGroup>>scanWith(
 					ArrayList::new,
 					(a, b) -> Stream.concat(a.stream(), Stream.of(b)).collect(Collectors.toList())
 			)
 			.lastOrError()
 			.map(particleGroups -> {
 				List<ParticleGroup> allParticleGroups = new ArrayList<>(particleGroups);
-				Atom atom = new Atom(particleGroups);
+				ImmutableMap<String, String> metaData = ImmutableMap.of(
+					Atom.METADATA_TIMESTAMP_KEY, String.valueOf(System.currentTimeMillis()));
```

Comment:
At some point in the future we will want to be using RTP time here, so it would be good to abstract this out into another method, perhaps in a new utility class or even an injectable `LongSupplier`.

Reply:
Cool, will do.